homeassistant/components/philips_air_purifier/fan.py

```diff
@@ -0,0 +1,166 @@
+"""The Purifier Fan Entity allows controlling a Philips Air Purifier's fan."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.fan import (
+    SUPPORT_PRESET_MODE,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import percentage_to_ordered_list_item
+
+from .client import FanSpeed, Mode, ReliableClient, Status
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a configured Air Purifier by creating the Fan entity."""
+    client = hass.data[DOMAIN][config_entry.entry_id]
+
+    initial_data = {
+        "unique_id": config_entry.unique_id,
+        "name": config_entry.title,
+        "model": config_entry.data["model"],
+    }
```

Comment:
Why not just pass the config entry to your entity? I think this will clean up your code.